src/translate_c.zig

```diff
@@ -1892,7 +1896,13 @@ fn isBoolRes(res: *ast.Node) bool {
         => return true,
 
         .GroupedExpression => return isBoolRes(@fieldParentPtr(ast.Node.GroupedExpression, "base", res).expr),
-
+        .If => {
+            const if_node = @fieldParentPtr(ast.Node.If, "base", res);
+            if (if_node.@"else") |else_node| {
+                return isBoolRes(if_node.body) and isBoolRes(else_node.body);
```

Comment:
This will not work if only one branch results in a boolean:
```c
int foo(int a) {
    return a ? a == 2 : 1;
}
```
I think the correct way to handle this would be to do the casting in `transBinaryConditionalOperator` and `transConditionalOperator`.

Reply:
Ok, I reverted this one; maybe this should be addressed in a separate PR since it's really a different problem? I also found another unrelated case where boolToInt isn't called (initializing global variables)

Reply:
I already fixed that in #7479 and I can fix this there too.